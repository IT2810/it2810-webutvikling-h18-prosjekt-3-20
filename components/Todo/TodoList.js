import React, { Component } from 'react';
import {
  FlatList,
} from 'react-native';
import {
  List,
} from 'react-native-elements';
import PropTypes from 'prop-types';
import TodoItem from './TodoItem';

export default class TodoList extends Component {
  static propTypes = {
    onCheckBoxPress: PropTypes.func.isRequired,
    onRemoveTodo: PropTypes.func.isRequired,
    todos: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
      date: PropTypes.string.isRequired,
      completed: PropTypes.bool.isRequired,
      coordinates: PropTypes.shape({
        latitude: PropTypes.number.isRequired,
        longitude: PropTypes.number.isRequired,
      }).isRequired,
    })).isRequired,
  };

  render() {
    // sort todolist on date so that the flat list is rendered in chronological order
    return (
      <List>
        <FlatList
          data={this.props.todos}
          extraData={this.props}
          renderItem={({ item }) => (
            <TodoItem
              item={item}
              onRemoveTodo={this.props.onRemoveTodo}
              onCheckBoxPress={this.props.onCheckBoxPress}
            />
          )}
          keyExtractor={(item, index) => index.toString()}
        />
      </List>
    );
  }
}
