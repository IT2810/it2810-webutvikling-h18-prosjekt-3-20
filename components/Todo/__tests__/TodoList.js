import { FlatList } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TodoList from '../TodoList';

const mockedTodos = [
  {
    name: 'Test 1',
    date: '2018-10-09 09:30:00',
    completed: false,
    coordinates: {
      latitude: 12.12401,
      longitude: 21.21324,
    },
  },
  {
    name: 'Test 2',
    date: '2018-10-10 10:25:25',
    completed: true,
    coordinates: {
      latitude: 12.12401,
      longitude: 21.21324,
    },
  },
];

describe('<TodoList/>', () => {
  it('should render empty component', () => {
    const tree = renderer.create(<TodoList todos={[]} onCheckBoxPress={jest.fn()}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render filled component', () => {
    const component = renderer.create(<TodoList todos={mockedTodos} onCheckBoxPress={jest.fn()}/>);

    const list = component.root.findByType(FlatList);

    expect(component.toJSON()).toMatchSnapshot();
    expect(list).toBeTruthy();
    expect(list.props.data.length).toBe(mockedTodos.length);
  });
  it('should pass correct number of todos to flatlist', () => {
    const wrapper = shallow(<TodoList todos={mockedTodos} onCheckBoxPress={jest.fn()}/>);
    const flatList = shallow(wrapper.find('FlatList').getElements()[0]);
    expect(flatList.instance().props.data.length).toEqual(mockedTodos.length);
  });
});
