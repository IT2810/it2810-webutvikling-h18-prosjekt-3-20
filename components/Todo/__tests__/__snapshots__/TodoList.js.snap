// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList/> should render empty component 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#bbb",
        "borderTopWidth": 1,
        "marginTop": 20,
      },
      undefined,
    ]
  }
/>
`;

exports[`<TodoList/> should render filled component 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#bbb",
        "borderTopWidth": 1,
        "marginTop": 20,
      },
      undefined,
    ]
  }
/>
`;
